genbtf.sh

```diff
@@ -64,48 +97,75 @@ cat <<EOF
 
 EOF
 
-Commit=$(cd $1 && git rev-parse HEAD)
-if [ -z $Commit ]; then
-	exit 1
-fi
-
 printf "const char *btfhub_archive_commit=\"%s\";\n\n" "$Commit"
 
-# Table of all successfull kernels, so we can create all_btfs[]
-typeset -a Good
-typeset -i Total=0
+function do_arch
+{
 
-for s in $Srcs; do
-	distro=${s%/*}
-	s="$1/$s"
-	for k in $(find $s -name '*.tar.xz'); do
-		btf=$(basename ${k%%.tar.xz})
-		name="$distro-${btf%%.btf}"
-		name=$(echo $name | tr \\055\\056\\057 _)
-		tar xf $k || die "oh noes"
-		if ./quark-btf -g $name -f $btf 2>/dev/null; then
-			echo "$name OK" 1>&2
-			Good+=("$name")
-		else
-			echo "$name FAIL" 1>&2
-		fi
-		rm -f $btf
-		Total=$((Total+1))
+	if [ $1 = amd64 ]; then
+		Arch=amd64
+		Srcs="$Srcs_amd64"
+	elif [ $1 = arm64 ]; then
+		Arch=aarch64
+		Srcs="$Srcs_arm64"
+	else
+		die "bad arch"
+	fi
+
+	printf "#ifdef __%s__\n\n" $Arch
```

Comment:
Do you need this? Or line 140?

Reply:
I do, this is purely to save space, basically an amd64 build will only compile the amd64 entries, and an arm64 only the arm64 one.